vortex-expr/src/display.rs

```diff
@@ -0,0 +1,150 @@
+use core::fmt;
+use std::fmt::{Display, Formatter};
+
+use vortex_dtype::{match_each_native_ptype, DType};
+use vortex_scalar::{BoolScalar, PrimitiveScalar};
+
+use crate::expressions::{ConjunctionExpr, DNFExpr, FieldExpr, PredicateExpr, Value};
+use crate::operators::Operator;
+use crate::scalar::ScalarDisplayWrapper;
+
+impl Display for DNFExpr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let iter = self.conjunctions.iter();
+        let mut first = true;
+        for conj in iter {
+            if first {
+                first = false;
+            } else {
+                write!(f, "\nOR \n")?;
+            }
+
+            write!(f, "{conj}")?;
+        }
+        Ok(())
+    }
+}
+
+impl Display for ConjunctionExpr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let iter = self.predicates.iter();
+        let mut first = true;
+        for disj in iter {
+            if first {
+                first = false;
+            } else {
+                write!(f, " AND ")?;
+            }
+
+            write!(f, "{disj}")?;
+        }
+        Ok(())
+    }
+}
+
+impl Display for PredicateExpr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "({} {} {})", self.left, self.op, self.right)
+    }
+}
+
+/// Alternative display for scalars
+impl Display for ScalarDisplayWrapper<'_> {
```

Comment:
Probably not worth having for now. If we do want it, should add it to the scalar crate maybe called `TypedScalarDisplay`